app/adapters/alma_adapter/alma_item.rb

```diff
@@ -7,6 +7,14 @@ def initialize(item)
       super(item)
     end
 
+    def self.reserve_location?(library_code, location_code)
+      return false if library_code.nil? || location_code.nil?
+      Rails.cache.fetch("library_#{library_code}_#{location_code}", expires_in: 30.minutes) do
```

Comment:
Notice that I am caching this Alma value for 30 minutes since we want to reduce our calls to Alma and these values don't change often.

Reply:
Also, notice that there are two "reserve" methods in this class, one is a class method and the other is an instance method. This is because we need this same logic from outside this class (i.e. in `alma_adapter/availability_status.rb`)